sources/osgAnimation/MorphGeometry.js

```diff
@@ -23,28 +23,37 @@ var MorphGeometry = function () {
 
     this._targets = []; // Target list (Geometry)
     this._stateSetAnimation = new StateSet(); // StateSet to handle morphAttribute
-    this._targetWeights = new Float32Array( 4 ); // Fixed length array feed by UpdateMorph
+    this._targetWeights = new Float32Array( MorphGeometry.MAX_MORPH_GPU ); // Fixed length array feed by UpdateMorph
 
+    this._morphAttribute = undefined;
     this._morphTargetNames = undefined;
-    this._updateMorph = undefined;
+
+    this._maxMorphGPU = MorphGeometry.MAX_MORPH_GPU; // used by updateMorph to limit the number of morphed attributes done by the gpu
 
     this._isInitialized = false;
 };
 
 // sync with UpdateMorph
 var EFFECTIVE_EPS = MorphGeometry.EFFECTIVE_EPS = 0.05;
 
+// this should be constant, if you change it only do it at parse time, otherwise it's better to call setMaximumPossibleMorphGPU
+MorphGeometry.MAX_MORPH_GPU = 4;
+
 MorphGeometry.prototype = MACROUTILS.objectLibraryClass( MACROUTILS.objectInherit( Geometry.prototype, {
 
     init: function () {
-        var animAttrib = new MorphAttribute( Math.min( 4, this.getMorphTargets().length ) );
-        this.getStateSetAnimation().setAttributeAndModes( animAttrib, StateAttribute.ON );
-        animAttrib.setTargetWeights( this.getTargetsWeight() );
+        if ( this._morphAttribute ) {
```

Comment:
Why ? there is a this._isInitialized = true at line 62 https://github.com/cedricpinson/osgjs/pull/749/files#diff-938f2ccc86a8b794c47cde2b61e74875R62